cmd/nvidia-ctk-installer/container/runtime/containerd/config_test.go

```diff
@@ -0,0 +1,1496 @@
+/*
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package containerd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	cli "github.com/urfave/cli/v3"
+
+	"github.com/NVIDIA/nvidia-container-toolkit/cmd/nvidia-ctk-installer/container"
+)
+
+// TestContainerdConfigLifecycle tests the complete Setup->Cleanup lifecycle for both v1 and v2 configs.
+func TestContainerdConfigLifecycle(t *testing.T) {
+	c := &cli.Command{
+		Name: "test",
+	}
+
+	testCases := []struct {
+		description string
+		// configVersion
+		// 0 for auto-detect
+		// 1 for v1
+		// 2 for v2
+		// 3 for v3
+		configVersion               int
+		containerOptions            container.Options
+		options                     Options
+		prepareEnvironment          func(*testing.T, string) error
+		expectedSetupError          error
+		assertSetupPostConditions   func(*testing.T, string) error
+		expectedCleanupError        error
+		assertCleanupPostConditions func(*testing.T, string) error
+	}{
+		// V1 test cases
+		// Note: We don't test "v1: top-level config does not exist" because new configs
+		// are always created as v2 configs by the containerd package
```

Comment:
Is this true even if the version is explicitly set to 1?

OK. The issue is that the `configVersion` in the `testCases` struct is never used when creating the config object. 

I think this is OK given the current behaviour, so let's remove the `configVersion` instead.